src/cargo/core/compiler/unit_dependencies.rs

```diff
@@ -168,7 +178,7 @@ fn deps_of_roots<'a, 'cfg>(roots: &[Unit<'a>], mut state: &mut State<'a, 'cfg>)
             } else if unit.target.is_custom_build() {
                 // This normally doesn't happen, except `clean` aggressively
                 // generates all units.
-                UnitFor::new_build()
+                UnitFor::new_build(false)
```

Comment:
To make sure I understand, this is where the "subtelty" of `build_dep` happens, right? I depend on the `host` version of a build script being compiled, but the compilation doesn't have the `build_dep` flag to `true`? (but transitive deps have it set to `true`).

I may also be getting mixed up again about what `is_custom_build` is again and which is "run the script" and which is "build the script"

Reply:
`is_custom_build` is true for both `CompileMode::RunCustomBuild` and `CompileMode::Build` of a build.rs script.  Normally code uses `is_run_custom_build` if it needs just one or the other.

Yea, any situation there is `UnitFor::new_build(false)` is the "subtle" difference.  That is, whenever adding a `build.rs` script for a normal dependency.

Reply:
Ok sounds good to me!